core/src/main/java/nu/yona/server/subscriptions/service/UserPrivateDto.java

```diff
@@ -34,30 +34,28 @@
 	private final UUID userAnonymizedId;
 	private final UUID namedMessageSourceId;
 	private final UUID anonymousMessageSourceId;
-	private final Set<UUID> buddyIds;
-	private final Function<Set<UUID>, Set<BuddyDto>> buddyIdToDtoMapper;
+	private final Set<BuddyDto> buddies;
 
 	@JsonCreator
 	public UserPrivateDto(@JsonProperty("nickname") String nickname)
 	{
-		this(Optional.empty(), null, nickname, null, null, Collections.emptySet(), Collections.emptySet(), null, null,
+		this(Optional.empty(), null, nickname, null, null, Collections.emptySet(), Collections.emptySet(), null,
 				new VPNProfileDto(null));
 	}
 
 	UserPrivateDto(Optional<LocalDate> lastMonitoredActivityDate, String yonaPassword, String nickname, UUID namedMessageSourceId,
-			UUID anonymousMessageSourceId, Set<GoalDto> goals, Set<UUID> buddyIds,
-			Function<Set<UUID>, Set<BuddyDto>> buddyIdToDtoMapper, UUID userAnonymizedId, VPNProfileDto vpnProfile)

```

Comment:
It was introduced for [YD-245](http://jira.yona.nu/browse/YD-245). See [this commit](https://github.com/yonadev/yona-server/pull/198/commits/da09c055159a728ff5b0ed4dc831581282d9eafd). I don't know the exact reason, because I was on holiday. The commit message reads: "UserDTO (with nested DTOs) now fetches the buddies when needed instead of having the controller set them explicitly".